cwms-data-api/src/main/java/cwms/cda/data/dao/LocationLevelsDaoImpl.java

```diff
@@ -186,50 +233,96 @@ public LocationLevels getLocationLevels(String cursor, int pageSize,
 
         usace.cwms.db.jooq.codegen.tables.AV_LOCATION_LEVEL view = AV_LOCATION_LEVEL;
         usace.cwms.db.jooq.codegen.tables.AV_VIRTUAL_LOCATION_LEVEL virtView = AV_VIRTUAL_LOCATION_LEVEL;
+        usace.cwms.db.jooq.codegen.tables.AV_LOC_ALIAS aliasView = AV_LOC_ALIAS;
+
+        Condition whereCondition;
+        Condition standardWhereCondition = (DSL.upper(view.UNIT_SYSTEM).eq(unit.toUpperCase()))
+            .or(virtView.LOCATION_LEVEL_ID.isNotNull());
+        LocationFieldMapping mapping;
+        if (includeAliases) {
+            mapping = new AliasedLocationFieldMapping();
+        } else {
+            mapping = new JooqLocationFieldMapping();
+        }
 
-        Condition whereCondition = (DSL.upper(view.UNIT_SYSTEM).eq(unit.toUpperCase()))
-                .or(virtView.LOCATION_LEVEL_ID.isNotNull());
+        whereCondition = (mapping.getUnitSystem(TABLE_ALIAS1, view).eq(unit.toUpperCase()))
```

Comment:
Feels weird to me that mapping.getUnitSystem needs to know the table_alias.  Can we build the mapping with the alias and the views so we don't have to pass it in?